feature-libs/asm/root/interceptors/user-id.interceptor.ts

```diff
@@ -0,0 +1,76 @@
+/*
+ * SPDX-FileCopyrightText: 2022 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  HttpEvent,
+  HttpHandler,
+  HttpInterceptor,
+  HttpRequest,
+} from '@angular/common/http';
+import { Inject, Injectable } from '@angular/core';
+import {
+  OCC_ASM_TOKEN,
+  OCC_USER_ID_CONSTANTS,
+  UserIdService,
+} from '@spartacus/core';
+import { Observable, of } from 'rxjs';
+import { concatMap, map } from 'rxjs/operators';
+
+/**
+ * Looks for a specific key in the HttpRequest's context (OCC_ASM_TOKEN) to decide when to
+ * configure a request with 'sap-commerce-cloud-user-id' header.
+ */
+@Injectable({ providedIn: 'root' })
+export class UserIdInterceptor implements HttpInterceptor {
```

Comment:
I think that precise naming improves the code readability, may I suggest we give the interceptor a name that communicates its purpose.  Perhaps `UserIdHeaderInterceptor` or `UserIdHttpHeaderInterceptor`